util/async_file_reader.cc

```diff
@@ -0,0 +1,74 @@
+//  Copyright (c) Meta Platforms, Inc. and its affiliates. All Rights Reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+//
+#if USE_COROUTINES
+#include "util/async_file_reader.h"
+
+namespace ROCKSDB_NAMESPACE {
+bool AsyncFileReader::MultiReadAsyncImpl(ReadAwaiter* awaiter) {
+  if (tail_) {
+    tail_->next_ = awaiter;
+  }
+  tail_ = awaiter;
+  if (!head_) {
+    head_ = awaiter;
+  }
+  num_reqs_ += awaiter->num_reqs_;
+  awaiter->io_handle_.resize(awaiter->num_reqs_);
+  awaiter->del_fn_.resize(awaiter->num_reqs_);
+  for (size_t i = 0; i < awaiter->num_reqs_; ++i) {
+    awaiter->io_handle_.push_back(nullptr);
+    awaiter->del_fn_.push_back(nullptr);
+    awaiter->file_
+        ->ReadAsync(
+            awaiter->read_reqs_[i], awaiter->opts_,
+            [](const FSReadRequest& req, void* cb_arg) {
+              FSReadRequest* read_req = static_cast<FSReadRequest*>(cb_arg);
+              read_req->status = req.status;
+              read_req->result = req.result;
+            },
+            &awaiter->read_reqs_[i], &awaiter->io_handle_[i],
+            &awaiter->del_fn_[i], Env::IOPriority::IO_TOTAL)
+        .PermitUncheckedError();
+  }
+  return true;
+}
+
+void AsyncFileReader::Poll() {
```

Comment:
I'm a little bit confused by the naming here (same as FileSytem::Poll()). This is called Poll(), but it is actually blocking until all the results return. Is it correct? If It is the case, it probably should be called something ...Wait..(), rather than Poll().

Reply:
Yes, its a blocking call. Wait() sounds like a better name to make the purpose clear. I'll change it.